src/content/docs/react/hooks.mdx

```diff
@@ -1,3 +1,412 @@
+  </TabItem>
+  <TabItem label="Usage">
+    <Code code={useEffectUsage} lang="jsx" title="Usage.jsx" />
+  </TabItem>
+  <TabItem label="Component">
+    <Code code={DataFetchingComponent} lang="jsx" title="DataFetchingComponent.jsx" />
+  </TabItem>
+</Tabs>
+
+In this example: The side effect is fetching data from an API when the component first renders. The [] (empty dependency array) ensures the effect only runs once, similar to `componentDidMount`.
+
+### Controlling using Dependency array
+
+You can control when an effect runs by adding dependencies. The effect will only run again if one of the dependencies changes.
+
+```jsx
+useEffect(() => {
+    // This effect runs when 'count' changes
+    console.log(`The count is now ${count}`);
+}, [count]); // Effect runs only when 'count' changes
+```
+
+### Cleaning up effects
+
+Sometimes, side effects like subscriptions or timers need to be cleaned up to prevent memory leaks. `useEffect` can return a cleanup function that runs when the component unmounts or before the effect re-runs.
+
+import Cleanup from './codes/hooksCode/Others/Cleanup.txt?raw';
+
+<Tabs>
+  <TabItem label="Usage">
+    <Code code={Cleanup} lang="jsx" title="CleanUp.jsx" />
+  </TabItem>
+</Tabs>
+
+### Additional Hooks
+
+
+### `useContext` : Accessing Global State with React’s Context API
+
+The useContext hook is used to access values provided by the Context API in React. The Context API allows you to create and manage global state, which can be shared across different components without passing props down through every level of the component tree. This is particularly useful for managing themes, authentication status, or user settings that are needed by many components
+
+useContext allows a component to consume the nearest Context Provider’s value without needing to pass props through intermediate components. This helps reduce "prop drilling" (passing props down multiple levels) and makes the code more readable.
+
+**Steps to use `useContext`**
+
+ - First, create a context using React.createContext(). This provides a default value for the context.
+
+ ```jsx
+ const MyContext = React.createContext(defaultValue);
+```
+
+ - Wrap your component tree with a Context Provider to supply a value that child components can consume.
+
+```jsx
+<MyContext.Provider value={someValue}>
+    <ChildComponent />
+</MyContext.Provider>
+```
+
+ - Use the useContext hook inside any component that needs the context value.
+
+ ```jsx
+ const value = useContext(MyContext);
+```
+
+### Example of `useContext`
+
+import useContextComponent from './codes/hooksCode/useContext/App.txt?raw';
+import useContextUsage from './codes/hooksCode/useContext/Usage.txt?raw';
+
+<Tabs>
+  <TabItem label="Dependencies">
+    Install the following libraries. Assuming you have installed
+    <Code code={`npm install react react-dom`} lang="shell" />
+  </TabItem>
+  <TabItem label="Usage">
+    <Code code={useContextUsage} lang="jsx" title="Usage.jsx" />
```

Comment:
Please know that usage means you have to give proper working code it is not something like syntax. for example:

if you have one more file that has code then in usage file you need to give code snippet of using that component code.